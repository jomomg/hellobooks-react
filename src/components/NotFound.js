import React from "react";
import {Route} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import ErrorIcon from "@material-ui/icons/Error";

/**
 * 404 page
 */
const NotFound = () => {
    return (
        <div style={{textAlign: "center"}}>
            <ErrorIcon style={{color: "red", height: 300, width: 300, margin: "auto"}} />
            <Typography variant="display3" style={{color: "white"}}>
            404
            </Typography>
            <Typography variant="display3" style={{color: "white"}}>
            Sorry, Not found
            </Typography>
            <Route render={({ history }) => (
                <Button
                    style={{backgroundColor: "orange"}}
                    onClick={()=>history.push("/")}>
                Go Home
                </Button>
            )}/>
        </div>
    );
};

export default NotFound;