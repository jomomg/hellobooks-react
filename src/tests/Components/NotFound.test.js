import React from "react";
import NotFound from "../../components/NotFound";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("<NotFound/>", () => {
    test("it renders successfully", () => {
        const notFound = shallow(<NotFound />);
        expect(shallowToJson(notFound)).toMatchSnapshot();
    });
    test("it redirects to home", () => {
        const notFound = shallow(<NotFound />);
        expect(notFound).toExist();
    });
});