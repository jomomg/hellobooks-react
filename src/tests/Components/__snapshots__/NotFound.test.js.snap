// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> it renders successfully 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <pure(Error)
    style={
      Object {
        "color": "red",
        "height": 300,
        "margin": "auto",
        "width": 300,
      }
    }
  />
  <WithStyles(Typography)
    style={
      Object {
        "color": "white",
      }
    }
    variant="display3"
  >
    404
  </WithStyles(Typography)>
  <WithStyles(Typography)
    style={
      Object {
        "color": "white",
      }
    }
    variant="display3"
  >
    Sorry, Not found
  </WithStyles(Typography)>
  <Route
    render={[Function]}
  />
</div>
`;
